querier/parser/parser.go

```
@@ -0,0 +1,266 @@
+
+	p.nextToken()
+
+	if !p.peekTokenTypeIs(token.INT) {
+		return fmt.Errorf("error when parsing `limit`: %w", p.createPeekError(token.INT))
+	}
+
+	p.nextToken()
+
+	limit, err := strconv.Atoi(p.curToken.Literal)
+	if err != nil {
+		return fmt.Errorf("cannot parse limit value: `%s` is not a valid integer.", p.curToken.Literal)
```

golangci-lint (2.11.4)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)